sdk/servicebus/service-bus/review/service-bus.api.md

```diff
@@ -52,10 +61,65 @@ export interface CreateBatchOptions extends OperationOptionsBase {
     maxSizeInBytes?: number;
 }
 
+// @public
+export interface CreateQueueOptions {
+    authorizationRules?: AuthorizationRule[];
+    autoDeleteOnIdle?: string;
+    deadLetteringOnMessageExpiration?: boolean;
+    defaultMessageTtl?: string;
+    duplicateDetectionHistoryTimeWindow?: string;
+    enableBatchedOperations?: boolean;
+    enablePartitioning?: boolean;
+    forwardDeadLetteredMessagesTo?: string;
+    forwardTo?: string;
+    lockDuration?: string;
+    maxDeliveryCount?: number;
+    maxSizeInMegabytes?: number;
+    name: string;
```

Comment:
Generally, any interface with `Options` in the name have been such that all properties in it are optional. We are now introducing a case where this is not so. For example, `name` in `CreateQueueOptions` and `CreateTopicOptions` is mandatory.

Should we instead change the `createQueue(CreateQueueOptions)` to `createQueue(name, CreateQueueOptions)` and remove the mandatory name from the options?

cc @JoshLove-msft

Reply:
Right now

```js
createQueue(queueName: string, operationOptions?: OperationOptions): Promise<QueueResponse>;
createQueue(queue: CreateQueueOptions, operationOptions?: OperationOptions): Promise<QueueResponse>;
```

May be this should be 
```js
// CreateQueueOptions extends OperationOptions
createQueue(queueName: string, options?: CreateQueueOptions): Promise<QueueResponse>;
```

cc @bterlson, @xirzec

Reply:
Hmm, one benefit of keeping the name in the options is it allows for creating a new CreateQueueOptions instance from an existing QueueProperties.

Reply:
you can still do that. Showing C# code below from https://github.com/Azure/azure-sdk-for-net/issues/13352
Instead of
``cs
QueueProperties queue = await client.GetQueueAsync(queueName);
var options = new CreateQueueOptions(queue);
options.Name = "new-queue";
QueueProperties newQueue = await client.CreateQueueAsync(options);
``
you will be doing
```cs
QueueProperties queue = await client.GetQueueAsync(queueName);
var options = new CreateQueueOptions(queue);
QueueProperties newQueue = await client.CreateQueueAsync("new-queue", options);
```

Reply:
> you can still do that. Showing C# code below from [Azure/azure-sdk-for-net#13352](https://github.com/Azure/azure-sdk-for-net/issues/13352)
> Instead of
> 
> ```cs
> QueueProperties queue = await client.GetQueueAsync(queueName);
> var options = new CreateQueueOptions(queue);
> options.Name = "new-queue";
> QueueProperties newQueue = await client.CreateQueueAsync(options);
> ```
> 
> you will be doing
> 
> ```cs
> QueueProperties queue = await client.GetQueueAsync(queueName);
> var options = new CreateQueueOptions(queue);
> QueueProperties newQueue = await client.CreateQueueAsync("new-queue", options);
> ```

Fair, I guess since we will generally be changing the name, it doesn't buy us much to copy it.

Reply:
Are you thinking of dropping down to a single method with an optional options?

Reply:
In JS, yes.

Reply:
I think this makes sense, though the .NET guidelines have an example using pattern currently implemented:
https://azure.github.io/azure-sdk/dotnet_introduction.html#dotnet-parameters
```c#
    // complex service method
    public virtual Response<BlobInfo> CreateBlob(BlobCreateOptions options = null, CancellationToken cancellationToken = default);

    // convinience overload[s]
    public virtual Response<BlobContainerInfo> CreateBlob(string blobName, CancellationToken cancellationToken = default);
```

Reply:
Updated as suggested  https://github.com/Azure/azure-sdk-for-js/pull/10049/commits/d5b1446fca43531f74834c262c1c1a62a62cddec

![image](https://user-images.githubusercontent.com/10452642/88351362-ded3a200-cd0a-11ea-93e9-75a6fb97e56f.png)